azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/algorithms/RsaSignature.java

```diff
@@ -10,209 +10,414 @@
+
+        return result;
+    }
 
     /*
      * See https://tools.ietf.org/html/rfc3447#section-5.2.1
      */
     protected BigInteger RSASP1(RSAPrivateKey K, BigInteger m) {
-    	
-    	if ( K == null ) {
-    		throw new IllegalArgumentException("K");
-    	}
-    	
-    	if ( m == null ) {
-    		throw new IllegalArgumentException("m");
-    	}
-    	
-    	BigInteger n = K.getModulus();
-    	BigInteger d = K.getPrivateExponent();
-    	
-    	if ( m.compareTo(BigInteger.ONE) == -1 || m.compareTo(n) != -1 ) {
-    		throw new IllegalArgumentException("message representative out of range");
-    	}
-    	
-    	return m.modPow(d, n);
+
+        if (K == null) {
+            throw new IllegalArgumentException("K");
+        }
+
+        if (m == null) {
+            throw new IllegalArgumentException("m");
+        }
+
+        BigInteger n = K.getModulus();
+        BigInteger d = K.getPrivateExponent();
+
+        if (m.compareTo(BigInteger.ONE) == -1 || m.compareTo(n) != -1) {
+            throw new IllegalArgumentException("message representative out of range");
+        }
+
+        return m.modPow(d, n);
     }
-    
+
     /*
      * See https://tools.ietf.org/html/rfc3447#section-5.2.2
      */
     protected BigInteger RSAVP1(RSAPublicKey K, BigInteger s) {
-    	
-    	if ( K == null ) {
-    		throw new IllegalArgumentException("K");
-    	}
-    	
-    	if ( s == null ) {
-    		throw new IllegalArgumentException("s");
-    	}
-    	BigInteger n = K.getModulus();
-    	BigInteger e = K.getPublicExponent();
-    	
-    	if ( s.compareTo(BigInteger.ONE) == -1 || s.compareTo(n) != -1 ) {
-    		throw new IllegalArgumentException("message representative out of range");
-    	}
-    	
-    	return s.modPow(e, n);
+
+        if (K == null) {
+            throw new IllegalArgumentException("K");
+        }
+
+        if (s == null) {
+            throw new IllegalArgumentException("s");
+        }
+        BigInteger n = K.getModulus();
+        BigInteger e = K.getPublicExponent();
+
+        if (s.compareTo(BigInteger.ONE) == -1 || s.compareTo(n) != -1) {
+            throw new IllegalArgumentException("message representative out of range");
+        }
+
+        return s.modPow(e, n);
+    }
+    
+    /**
+     * See https://tools.ietf.org/html/rfc3447#section-9.1.1
+     * 
+     * @param mHash the hashed message (hashed with the corresponding digest: i.e., SHA-256 for PS256, SHA-384 for PS384, SHA-512 for PS512 
+     * @param emBits the maximal bit length of the integer OS2IP(EM)
+     * @param messageDigest the messageDigest (or hash) that corresponds to the algorithm
+     * @return
+     */
+    protected byte[] EMSA_PSS_ENCODE_HASH(byte[] mHash, int emBits, MessageDigest messageDigest) {
+
+        // Let mHash = Hash(M), an octet string of length hLen.
+        // function takes in the mHash.
+
+        // salt length for the service corresponds to the digest length, which is the hash length / 8
```

Comment:
> which is the hash length / 8 [](start = 72, length = 29)

perhaps you should call this hash length in bytes as it might be confused with mHash.length() / 8